### ISSUE TITLE
Add hooks for system "cleanup" tasks

### ISSUE THREAD

**Issue by hadess** [hughsie/PackageKit#211]
It would be useful to be able to tack some tasks to the end of offline package updates, as "cleanup" or "optimisation" tasks.

In no particular order, those could be:
- RPM database cleanup
- logs rotation and compression
- old files trimming in /var/tmp
- sanity checking of old coredumps in abrt, removing the unreportable ones
- SSD filesystem trimming

The latter is the main reason why I'm interested in this feature. Trimming a filesystem lengthens the lifetime of an SSD, but might need tens of seconds or minutes to run, making the disk completely unavailable, so not something that can be easily run on boot, shutdown and even less while the machine is in use.

We cannot add this functionality as a separate service because:
```
<zbyszek> the upgrade service (e.g. packagekit-offline-update.service) schedules an irreversible reboot.target/start job, which replaces the old transaction, and systemd will kick out any jobs which were necessary for the old target but not the new one.
```
and running it as part of the shutdown process means no visual feedback, and a sub-par UI.

In short:
- add a way to tell PK to run some tasks before calling "reboot"
- make it possible for those tasks to report progress

cc @chergert

**Comment by hughsie** [comment]
Isn't this bigger than PK tho? I mean, we're going to be doing something related for rpm-ostreed on a Fedora atomic workstation, and that's PK-less.

**Comment by hadess** [comment]
PK is the one talking to the UI to report progress during updates, PK is the service that's launched for the `system-update.target`. When we have something else taking its place, I'm sure we'll be looking into what's needed to avoid regressions.

**Comment by chergert** [comment]
Another question is if we should only be doing this for file-systems effected by the installation or all file-systems including the user's home directory. My default Fedora installation (rawhide) seems to support fstrim out-of-the-box or at least `fstrim` didn't complain to me. So that is better than we had a year ago.

**Comment by hadess** [comment]
How it runs fstrim and for what is not relevant (yet). We need to be able to run it before actually doing it...
